### DISCUSSION TITLE
rAudio 20221007: Library Update complete after reboot only

### DISCUSSION THREAD

**Discussion by klassikmann** [rern/rAudio#1050] [Bug Reports]
Performing a local update in the library view results in a "Done" message, but the new content is visible in the GUI after a reboot only.

UPDATE:
The problem happened exactly once. I will continue to monitor and report the behavior of the new release.

**Comment by ceejayemm** [comment]
I am getting apparently the same issue. After the update completed the system rebooted (without any action from me) but the GUI was not available to me by access via SSH was.  I rebooted again, via SSH, but still nothing.  Journalctl -xe reports php errors:

Oct 07 18:31:22 RAOffice nginx[327]: 2022/10/07 18:31:22 [error] 327#327: *5 FastCGI sent in stderr: "PHP message: PHP Warning:  scan>
Oct 07 18:31:22 RAOffice nginx[327]: Stack trace:
Oct 07 18:31:22 RAOffice nginx[327]: #0 /srv/http/index-body.php(69): array_slice()
Oct 07 18:31:22 RAOffice nginx[327]: #1 /srv/http/index.php(30): include('/srv/http/index...')
Oct 07 18:31:22 RAOffice nginx[327]: #2 {main}
Oct 07 18:31:22 RAOffice nginx[327]:   thrown in /srv/http/index-body.php on line 69" while reading response header from upstream, cl>
Oct 07 18:31:25 RAOffice nginx[327]: 2022/10/07 18:31:25 [error] 327#327: *1 FastCGI sent in stderr: "PHP message: PHP Warning:  scan>
Oct 07 18:31:25 RAOffice nginx[327]: Stack trace:
Oct 07 18:31:25 RAOffice nginx[327]: #0 /srv/http/index-body.php(69): array_slice()
Oct 07 18:31:25 RAOffice nginx[327]: #1 /srv/http/index.php(30): include('/srv/http/index...')
Oct 07 18:31:25 RAOffice nginx[327]: #2 {main}
Oct 07 18:31:25 RAOffice nginx[327]:   thrown in /srv/http/index-body.php on line 69" while reading response header from upstream, cl>
lines 1866-1892/1892 (END)

the lines are too wide to screenshot so I captured the journalctl output to a file and attached it here:

[journalctl.txt](https://github.com/rern/rAudio-1/files/9736045/journalctl.txt)

Chris

**Reply by rern** [reply]
Hi Chris,

It might be some conflicts with the new feature, Server rAudio,
Try:
```sh
dir=/srv/http/shareddata
dirshareddata=/mnt/MPD/NAS/data
filesharedip=$dirshareddata/sharedip
echo data > /mnt/MPD/NAS/.mpdignore
mkdir -p $dirshareddata
mv $dir/iplist > $filesharedip
chmod 777 $filesharedip
umount -l $dir
sed -i "s|$dir|$dirshareddata|" /etc/fstab
systemctl daemon-reload
mount $dirshareddata
rmdir $dir
```

By the way, please try [Server rAudio](https://github.com/rern/rAudio-1/discussions/1053#discussion-4455633).

**Comment by klassikmann** [comment]
After some tests: The problem I described in my original post appeared exactly once, and that, immediately after the update.

**Reply by ceejayemm** [reply]
@rern 

I think your script needs a bit of work:

RAOffice:~  #  dir=/srv/http/shareddata
dirshareddata=/mnt/MPD/NAS/data
filesharedip=$dirshareddata/sharedip
echo data > /mnt/MPD/NAS/.mpdignore
mkdir -p $dirshareddata
mv $dir/iplist > $filesharedip
chmod 777 $filesharedip
umount -l $dir
sed -i "s|$dir|$dirshareddata|" /etc/fstab
systemctl daemon-reload
mount $dirshareddata
rmdir $dir

mv: missing destination file operand after '/srv/http/shareddata/iplist'
Try 'mv --help' for more information.
umount: /srv/http/shareddata: no mount point specified.
mount.nfs: mounting 192.168.86.250:/export/music/database/RAdb failed, reason given by server: No such file or directory
rmdir: failed to remove '/srv/http/shareddata': No such file or directory

Chris

**Reply by ceejayemm** [reply]
Ok, fixed primary problem at line 6, 

mv $dir/iplist > $filesharedip

should be:

mv $dir/iplist $filesharedip

Having corrected this I have been able to run remainder of script. However even having done this there are still PHP errors being listed in the journalctl -xe output.  I have attached most recent journalctl output here, errors start at Oct 08 17:34:20:

[journalctl.log](https://github.com/rern/rAudio-1/files/9739819/journalctl.log)

Chris

**Reply by ceejayemm** [reply]
@rern 

Reflashed a new SD card using i20220904 and upgraded to current (20221007) version, this has same 'mv' command error are reported above (mv $dir/iplist > $filesharedip should be: mv $dir/iplist $filesharedip) as shown here:

curl -sfLO https://github.com/rern/rAudio-1/raw/main/install.sh
chmod 755 install.sh
./install.sh "r1
Update
20221007"
mv: missing destination file operand after '/srv/http/shareddata/iplist'

Try 'mv --help' for more information.

cat: 

/srv/http/data/mpd/counts

: No such file or directory

Applying the current UPDATE branch seems to fix the 'mv' error:

curl -sfLO https://github.com/rern/rAudio-1/raw/UPDATE/install.sh
chmod 755 install.sh
./install.sh "r1
Update
UPDATE"
cat: 

/srv/http/data/mpd/counts: No such file or directory

but there is a problem, in both cases, with the 'counts' directory and still the  original errors with index.php / index-body.php so the system fails to display Playback screen, just shows a black screen instead.  Using http://raoffice/settings.php?p=system gets me to the System (and other configuration screens) but the Cog menu is unavailable as is the Library and Playlist screens.  

It seems I currently have a wounded system.  I am little surprised nobody else is reporting this as this is now a fairly bog-standard installation.

Chris

**Comment by ceejayemm** [comment]
@rern 

Whilst working on #1057 I think I now know the cause of the problems I have been having here.  I think the problems were caused by me still having the original 'Shared Data' options enabled whilst applying the 20221007 updates and this update then tried to install the 'new' share data processes which caused a conflict. 

I have now resolved this as shown in #1057 and I am back to things working (nearly) normally as I get to fully understand the changes made to the Shared Data setup.

Thanks

Chris
